button-merchant/src/androidTest/java/com/usebutton/merchant/TestManagerTest.java

```diff
@@ -0,0 +1,159 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.usebutton.merchant;
+
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.RETURNS_MOCKS;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+public class TestManagerTest {
+
+    private Context context;
+    private ButtonRepository buttonRepository;
+    private TestManager.Terminator terminator;
+    private TestManager testManager;
+    private ArgumentCaptor<Intent> intentCaptor;
+
+    @Before
+    public void setUp() {
+        context = mock(Context.class, RETURNS_MOCKS);
+        buttonRepository = mock(ButtonRepository.class);
+        terminator = mock(TestManager.Terminator.class);
+        testManager = new TestManager(context, buttonRepository, terminator);
+        intentCaptor = ArgumentCaptor.forClass(Intent.class);
+    }
+
+    @Test
+    public void parseIntent_unsupported_shouldNotSendResponse() {
+        Intent intent = createIntentForAction("unknown");
+
+        testManager.parseIntent(intent);
+
+        verifyZeroInteractions(context);
+    }
+
+    @Test
+    public void parseIntent_quit_harnessExists_shouldQuitApp() {
+        Intent intent = createIntentForAction(TestManager.ACTION_QUIT);
+
+        testManager.parseIntent(intent);
+
+        verifyBaseResponseStructure(TestManager.ACTION_QUIT);
+        verify(terminator).terminate();
+    }
+
+    @Test
+    public void parseIntent_quit_harnessDoesNotExist_shouldNotQuitApp() {
+        Intent intent = createIntentForAction(TestManager.ACTION_QUIT);
+        doThrow(ActivityNotFoundException.class).when(context).startActivity(any(Intent.class));
+
+        testManager.parseIntent(intent);
+
+        verify(terminator, never()).terminate();
+    }
+
+    @Test
+    public void parseIntent_getToken_shouldSendTokenInResponse() {
+        Intent intent = createIntentForAction(TestManager.ACTION_GET_TOKEN);
+        when(buttonRepository.getSourceToken()).thenReturn("test-token");
+
+        testManager.parseIntent(intent);
+
+        Uri data = verifyBaseResponseStructure(TestManager.ACTION_GET_TOKEN);
+        assertEquals("test-token", data.getQueryParameter("btn_ref"));
+    }
+
+    @Test
+    public void parseIntent_postInstall_shouldSendResultInResponse() {
```

Comment:
I also tested that we get a `false` case if post install didn't get called

Reply:
It's a primitive value so it'd be false anyway by default.